src/preprocessing/geometries/geometries.jl

```diff
@@ -55,3 +55,75 @@ function Base.intersect(initial_condition::InitialCondition,
 
     return intersect(result, Base.tail(geometries)...)
 end
+
+# This method is used in `boundary_zone.jl` and is defined here
+# to avoid circular dependencies with `TriangleMesh`
+"""
+    planar_geometry_to_face(planar_geometry::TriangleMesh)
+
+Extracts a simplified rectangular face and its normal vector from an arbitrary planar geometry
+(`TriangleMesh` loaded via [`load_geometry`](@ref))
+for use as a boundary zone interface in [`BoundaryZone`](@ref).
+This function computes the corner points of an oriented bounding box
+that best represents the essential orientation and extent of the input geometry.
+The geometry must be planar (all vertices should lie in the same plane),
+but can have complex or non-rectangular boundaries.
+
+!!! note "Face Normal Orientation"
+    All face normals of the input geometry must point inside the fluid domain.
+    The returned plane normal is computed by averaging all face normals, so consistent orientation is required.
+
+# Arguments
+- `planar_geometry`: A planar geometry (`TriangleMesh` loaded via [`load_geometry`](@ref)).
+
+# Returns
+- `face_vertices`: Tuple of three vertices defining the rectangular face (corner points of the oriented bounding box).
+- `face_normal`: Normalized normal vector of the face.
+
+# Example
+```jldoctest; output=false, filter=r"face = .*"
+file = pkgdir(TrixiParticles, "test", "preprocessing", "data")
+planar_geometry = load_geometry(joinpath(file, "inflow_geometry.stl"))
+
+face, face_normal = planar_geometry_to_face(planar_geometry)
+
+# output
+(face = ...)
+```
+"""
+function planar_geometry_to_face(planar_geometry::TriangleMesh)
+    face_normal = normalize(sum(planar_geometry.face_normals) / nfaces(planar_geometry))
+
+    face_vertices = oriented_bounding_box(stack(planar_geometry.vertices))
+
+    # Vectors spanning the face
+    edge1 = face_vertices[:, 2] - face_vertices[:, 1]
+    edge2 = face_vertices[:, 3] - face_vertices[:, 1]
+
+    if !isapprox(abs.(normalize(cross(edge2, edge1))), abs.(face_normal), atol=1e-2)
+        throw(ArgumentError("geometry might be not planar"))
+    end
```

Comment:
1. This is a weird check. You take the cross product of the edges and then make all coordinates positive and compare with the normal with all coordinates positive?
2. The error is also weird. What do you mean "might"? Either you confirmed it's not planar and throw an error, or you're not sure and show a warning.

```suggestion
    if abs(dot(edge1, face_normal)) > 1e-2 || abs(dot(edge2, face_normal)) > 1e-2
        throw(ArgumentError("geometry is not planar"))
    end
```